SpatialGDK/Source/SpatialGDK/Public/Schema/VirtualWorkerTranslation.h

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/Component.h"
+#include "SpatialCommonTypes.h"
+#include "Utils/SchemaUtils.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+
+namespace SpatialGDK
+{
+struct VirtualWorkerInfo
+{
+	VirtualWorkerId VirtualWorkerId;
+	FString PhysicalWorkerName;
+	Worker_EntityId ServerWorkerEntity;
+	Worker_PartitionId PartitionId;
+	Worker_EntityId ServerSystemWorkerEntity;
+};
+
+// The SpatialDebugging component exists to hold information which needs to be displayed by the
+// SpatialDebugger on clients but which would not normally be available to clients.
+struct VirtualWorkerTranslation : AbstractMutableComponent
+{
+	static const Worker_ComponentId ComponentId = SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID;
+
+	VirtualWorkerTranslation()
+		: VirtualWorkerMapping({})
+	{
+	}
+
+	VirtualWorkerTranslation(const TArray<VirtualWorkerInfo>& InVirtualWorkerMapping) { VirtualWorkerMapping = InVirtualWorkerMapping; }
+
+	VirtualWorkerTranslation(const Worker_ComponentData& Data)
+		: VirtualWorkerTranslation(Data.schema_type)
+	{
+	}
+
+	VirtualWorkerTranslation(Schema_ComponentData* Data)
+	{
+		Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data);
+
+		// Resize the map to accept the new data.
+		const uint32 VirtualWorkerCount = Schema_GetObjectCount(ComponentObject, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_MAPPING_ID);
+		VirtualWorkerMapping.Empty(VirtualWorkerCount);
+
+		for (uint32 i = 0; i < VirtualWorkerCount; i++)
+		{
+			// Get each entry of the list and then unpack the virtual and physical IDs from the entry.
+			Schema_Object* MappingObject = Schema_IndexObject(ComponentObject, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_MAPPING_ID, i);
+			const VirtualWorkerId VirtualWorkerId = Schema_GetUint32(MappingObject, SpatialConstants::MAPPING_VIRTUAL_WORKER_ID);
+			const PhysicalWorkerName WorkerName =
+				SpatialGDK::GetStringFromSchema(MappingObject, SpatialConstants::MAPPING_PHYSICAL_WORKER_NAME_ID);
+			const Worker_EntityId ServerWorkerEntityId =
+				Schema_GetEntityId(MappingObject, SpatialConstants::MAPPING_SERVER_WORKER_ENTITY_ID);
+			const Worker_PartitionId PartitionEntityId = Schema_GetEntityId(MappingObject, SpatialConstants::MAPPING_PARTITION_ID);
+			const Worker_EntityId SystemWorkerEntityId =
+				Schema_GetEntityId(MappingObject, SpatialConstants::MAPPING_SYSTEM_WORKER_ENTITY_ID);
+
+			VirtualWorkerMapping.Emplace(
+				VirtualWorkerInfo{ VirtualWorkerId, WorkerName, ServerWorkerEntityId, PartitionEntityId, SystemWorkerEntityId });
+		}
+	}
+
+	VirtualWorkerTranslation(const Worker_ComponentUpdate& Update)
```

Comment:
Allowing a constructor take an Update (instead of data) seems dangerous, as it might be incomplete. Do we actually need this?